src/line.rs

```diff
@@ -96,32 +97,102 @@ impl<'input, Endian> From<&'input [u8]> for DebugLine<'input, Endian>
     }
 }
 
+/// Shutup
+pub trait LineNumberProgramHeaderHolder<'input, Endian> : Deref<Target = LineNumberProgramHeader<'input, Endian>>
```

Comment:
Its generally frowned upon to `Deref` but *also* add methods. Generally a thing should be a magic pointer and only deref, or if it must also have its own methods, then it should use `As[Mut]Ref` instead of `Deref`. The reasoning is that code becomes harder to read with the "fake inheritence" that `Deref` provides, and method resolution is obscured as well.

So I would prefer:

```rust
pub trait LineNumberProgramHeaderHolder<'input, Endian> : AsRef<LineNumberProgramHeader<'input, Endian>>
```